subsys/bluetooth/services/fast_pair/fp_activation.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/atomic.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(fast_pair, CONFIG_BT_FAST_PAIR_LOG_LEVEL);
+
+#include <bluetooth/services/fast_pair.h>
+#include "fp_advertising.h"
+#include "fp_keys.h"
+#include "fp_auth.h"
+#include "fp_battery.h"
+#include "fp_registration_data.h"
+#include "fp_gatt_service.h"
+
+/* Enum used to describe Fast Pair state. */
+enum state {
+	STATE_DISABLED,
+	STATE_ENABLED,
+	STATE_ERROR,
+
+	STATE_COUNT,
+};
+
+static atomic_t state = ATOMIC_INIT(STATE_DISABLED);
+
+static int modules_init(void)
+{
+	int err;
+
+	err = fp_reg_data_init();
```

Comment:
I see that you are initializing and deinitializing modules in the same order and there are quite few of them.
Are there dependencies between those modules? 
Do the init functions need to be called in specific order?

Maybe we should deinitialize in inverted order to what we have in initialization?

Consider (if we have doubts that in the future particular order might be required then ignore it):
If we could initialize those modules in ANY order, maybe we could use iterable sections to register init/deinit callbacks in each module and just iterate through them here? Something like it is done in mcumgr (`MCUMGR_HANDLER_DEFINE`).
If we have to initialize in PARTICULAR order that current solution is good for me.

Reply:
I will implement the section approach.